syntax = "proto3";

package darknetServer;

service ImageDetection {
	rpc RequestDetection (KeyFrame) returns (DetectedObjects) {};
}

message KeyFrame {
	int32 width=1;
	int32 height=2;
	int32 numChannels=3;
	int32 widthStep = 4;
	repeated float data=5;
}


message DetectedObjects {
	message DetectedObject {
		message box {
			float x=1;
			float y=2;
			float w=3;
			float h=4;
		}

		box bbox=1;
		int32 classes=2;
		repeated float prob=3;
		repeated float mask=4;
		float objectness=5;
		int32 sort_class=6;
	}

	int32 numObjects=1;
	repeated DetectedObject objects=2;
}
